aiohttp/client.py

```diff
@@ -353,10 +359,58 @@ def _update_cookies(self, cookies):
             if isinstance(value, http.cookies.Morsel):
                 # use dict method because SimpleCookie class modifies value
                 # before Python 3.4
+                domain = value["domain"]
+                if domain.startswith("."):
+                    value["domain"] = domain[1:]
                 dict.__setitem__(self.cookies, name, value)
             else:
                 self.cookies[name] = value
 
+    def _filter_cookies(self, url):
+        """Returns this session's cookies filtered by their attributes"""
+        # TODO: filter by 'expires', 'path', ...
+        netloc = urllib.parse.urlsplit(url).netloc
+        is_ip = IPV4_PATTERN.match(netloc) is not None
```

Comment:
`ipaddress.IPAddress(netloc) ` to support both IPv4 and IPv6 in the right way without regexps (which accepts invalid IPv4 addresses).